app/actions.ts

```
@@ -1,321 +1,451 @@
+
+  let progressTracker = {
+    name: false,
+    description: false,
+    website: false,
+    objectives: false,
+  };
+
+  const responseCache = new Set(); // To track recent user responses
+
+  const updateProgressTracker = (field: string) => {
+    if (progressTracker.hasOwnProperty(field)) progressTracker[field] = true;
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)